src/main/java/hudson/remoting/Channel.java

```diff
@@ -1226,9 +1229,14 @@ public void dumpPerformanceCounters(PrintWriter w) throws IOException {
      * Number of RPC calls (e.g., method call through a {@linkplain RemoteInvocationHandler proxy})
      * that the other side has made to us but not yet returned yet.
      *
-     * @since 2.26.3
+     * @param w Output destination
+     * @throws IOException Error while creating or writing the channel information
+     * 
+     * @since 2.62.3 - stable 2.x (restricted)
+     * @since TODO 3.x - public version 
      */
-    public void dumpDiagnostics(PrintWriter w) throws IOException {
+    @Restricted(NoExternalUse.class)
+    public void dumpDiagnostics(@Nonnull PrintWriter w) throws IOException {
```

Comment:
Maybe IOException is YAGNI, but I would like to keep it since one may override Channel implementation and start throwing something. The current implementation does not throw exceptions